middleware/authz.js

```diff
@@ -0,0 +1,22 @@
+require("express-async-errors");
+const jwt = require("jsonwebtoken");
+
+module.exports =  (listOfRolesAllowed) => {
+  return async (req, res, next) => {
+    const token = req.header("x-auth-token");
+    if (!token)
+      return res.status(401).send(`Access denied. No token provided.`);
+
+    const decoded = jwt.verify(token, process.env.JWT_PRIVATE_KEY);
+    const isRoleAllowed = listOfRolesAllowed.find((role) =>
```

Comment:
Given that you only have two roles, you can simplify the logic by passing the role directly as an argument to the middleware. This approach eliminates the need for an array and allows you to handle role-specific logic more directly, making the code cleaner and easier to understand.